test/connector/http/generic/http_test.go

```diff
@@ -31,21 +32,16 @@ func TestCreds(t *testing.T) {
 	for _, testCase := range testCases {
 		t.Run(testCase.description, func(t *testing.T) {
 			srv, err := httpServer(testCase.serverUsername, testCase.serverPassword)
-			if !assert.NoError(t, err) {
-				return
-			}
+			assert.NoError(t, err)
```

Comment:
The removal of the `if` is not correct. If the assertion fails, it will not immediately return but will continue running and likely cause a segfault. Same issue may be present in the other removals.